src/libstd/sys/windows/fs2.rs

```diff
@@ -323,6 +346,19 @@ impl FilePermissions {
     }
 }
 
+impl FileType {
+    pub fn is_dir(&self) -> bool {
+        !self.is_symlink() &&
+            self.file_attributes & c::FILE_ATTRIBUTE_DIRECTORY != 0
+    }
+    pub fn is_file(&self) -> bool {
+        !self.is_symlink() && !self.is_dir()
+    }
+    pub fn is_symlink(&self) -> bool {
+        self.file_attributes & libc::FILE_ATTRIBUTE_REPARSE_POINT != 0
+    }
```

Comment:
Not all [reparse points](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365503%28v=vs.85%29.aspx) are symlinks on Windows. Reparse points are a mechanism for storing some application defined data, that can be interpreted by a driver to act as a virtual filesystem at that point in the hierarchy, and are used to provide not only symlinks, but also mountpoints, deduplication, etc. See the [list of Microsoft defined reparse point tags](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365511%28v=vs.85%29.aspx) for details. It looks like you need to get the reparse point tag and check whether it is `IO_REPARSE_TAG_SYMLINK`. It looks only the `FindFirstFile`/`FindNextFile` family of functions return the [`WIN32_FIND_DATA`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365740%28v=vs.85%29.aspx) that contains this information. I think the best option would be to extend the `FileType` struct with something like `reparse_point_tag: Option<libc::DWORD>`, and when constructing it in places where you have that information; that is, in `DirEntry` where we get it from the `WIN32_FIND_DATAW`, and probably also change `lstat` to use `FindFirstFileW` so we can get this information in that call as well.